server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -0,0 +1,169 @@
+package terraform
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/activities"
+	"go.temporal.io/sdk/workflow"
+	"time"
+)
+
+const TaskQueue = "terraform"
+
+type PlanStatus int
+
+const (
+	Approved PlanStatus = iota
+	Rejected
+)
+
+func Workflow(ctx workflow.Context, request Request) error {
+	// TODO: should we set up a heartbeat? How long it takes to make progress on a terraform op activity is dependent on
+	// how complex of a change it runs
+	options := workflow.ActivityOptions{
+		TaskQueue:              TaskQueue,
+		ScheduleToCloseTimeout: 30 * time.Minute,
+		//HeartbeatTimeout:       2 * time.Minute,
+	}
+	ctx = workflow.WithActivityOptions(ctx, options)
+
+	sessionOptions := &workflow.SessionOptions{
+		CreationTimeout:  time.Minute,
+		ExecutionTimeout: time.Minute,
+	}
+	ctx, err := workflow.CreateSession(ctx, sessionOptions)
+	if err != nil {
+		return err
+	}
+	defer workflow.CompleteSession(ctx)
+
+	runner := newRunner(ctx, request)
+
+	// blocking call
+	return runner.Run(ctx)
+}
+
+type workerActivities interface {
+	GithubRepoClone(context.Context, activities.GithubRepoCloneRequest) error
+	TerraformInit(context.Context, activities.TerraformInitRequest) error
+	TerraformPlan(context.Context, activities.TerraformPlanRequest) error
+	TerraformApply(context.Context, activities.TerraformApplyRequest) error
+	ExecuteCommand(context.Context, activities.ExecuteCommandRequest) error
+	Notify(context.Context, activities.NotifyRequest) error
+	Cleanup(context.Context, activities.CleanupRequest) error
+}
+
+type Runner struct {
+	workerActivities
+	request Request
+}
+
+func newRunner(ctx workflow.Context, request Request) *Runner {
+	return &Runner{
+		workerActivities: activities.Terraform{},
+		request:          request,
+	}
+}
+
+func (r *Runner) Run(ctx workflow.Context) error {
+	// Clone repository into disk
+	err := workflow.ExecuteActivity(ctx, r.GithubRepoClone, activities.GithubRepoCloneRequest{}).Get(ctx, nil)
```

Comment:
It would be great to have this such that we're only keeping the root directory on disk, instead of cloning the whole repo.  Even if we clone the whole repo, and extract the root directory and delete the rest, that would be better.

Reply:
Will look into it! on success criteria for PR, I basically want this skeleton passed without the implementation details of each activity and the initial push event trigger. I'll update those in separate PRs.